src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,102 @@
+package duke;
+import java.io.IOException;
+
+
+public class Duke {
+    private TaskList taskList;
+    private UI userInterface;
+    
+    public Duke() {
+        try {
+            this.userInterface = new UI();
+            this.taskList = new TaskList(Storage.loadFile());
+        } catch (IOException e) {
+            System.out.println("An error has occurred: " + e);
+        }
+    }
+
+    public static void main(String[] args) throws IOException {
+        Duke duke = new Duke();
+        duke.run();
+    }
+    
+    public void run() {
+        userInterface.greetUser();
+        do{
+            try {
+                String input = Parser.getUserCommand();
+                String command = input.toLowerCase();
+
+                if (command.equals("bye")) {
+
+                    Storage.writeToFile(taskList);
+                    userInterface.goodbye();
+                    break;
+
+                } else if (command.equals("list")) {
+
+                    userInterface.listTasks(taskList);
+
+                } else if (command.split("\\s+")[0].equals("done")) {
+                    Task task = taskList.get(Integer.parseInt(command.split("\\s+")[1]) - 1);
+                    task.setDone();
+                    userInterface.taskCompletedMessage(task);
+                }else if (command.split("\\s+")[0].equals("delete")){
+                    if (command.split("\\s+").length != 2) {
+                        throw new IllegalUserInputException("PLease specify the correct argument number");
+                    }
+                    try {
+                        int i = Integer.parseInt(command.split("\\s+")[1]);
+                        Task.decrementTask();
+                        Task task = taskList.get(i-1);
+                        taskList.deleteTaskIndex(i-1);
+                        userInterface.taskDeletedMessage(task);
+                    } catch (TaskListError e) {
+                        System.out.println(e.getDetails());
+                        continue;
+                    }
+
+                } else {
+                    storeInput(input, taskList, userInterface);
+                }
+            } catch (IllegalArgumentException | IOException e) {
+                System.out.println(e);
+                continue;
+            }
+        } while (true);
```

Comment:
I am not too sure if the usage of while loop here is an acceptable standard, perhaps you can shift this to the top?